editoast/src/views/authz.rs

```diff
@@ -53,6 +56,53 @@ enum NoSuchUserError {
     NoSuchUser { user_id: i64 },
 }
 
+#[derive(Debug, serde::Serialize, utoipa::ToSchema)]
+#[cfg_attr(test, derive(serde::Deserialize, PartialEq))]
+struct WhoamiResponse {
+    id: i64,
+    name: String,
+    roles: Vec<Role>,
+}
+
+#[utoipa::path(
+    get,
+    path = "",
+    tag = "authz",
+    responses((
+        status = 200,
+        description = "Get the info of the current user",
+        body = inline(WhoamiResponse),
+    ))
+)]
+async fn whoami(
+    State(AppState { config, .. }): State<AppState>,
+    Extension(auth): AuthenticationExt,
+) -> Result<Json<WhoamiResponse>> {
+    // This is especially necessary as this endpoint is always queried by the frontend
+    // when loading. Making it return 401 results in a blank page.
+    // We return `Admin` as when no authorization is enabled, we want everyone to have
+    // access to full feature set of OSRD.
+    if !config.enable_authorization {
```

Comment:
We could handle this in the `Authorizer`. The `SkipAuthorization` could contain username, identity information. 

Here a suggestion of new workflow:
1. Try retrieve user information using regulator
2. If we must skip authorization then use `SkipAuthorization{user_identity, id, name}` variant
    - If no information retrieve then provide default values (like you're doing in this endpoint)
3. Otherwise create an `Authenticate` variant with all information
    - If no information retrieve then fail with a 403 (current behavior)

Reply:
Not sure what you mean. `authentication == SkipAuthorization` only happens in two cases:

- the header was put there by Core to bypass the authorization, and Core will never call `/authz/me`. This header is also blocked by the gateway, so you can't put it there using curl unless querying editoast directly.
- the authorization is disabled using the server config

In the latter, I'm not sure what the benefit is, especially now that we have impersonation?

Reply:
I did that, not sure if that's what you meant, but I think the operations order is a bit more clear like this anyway: d17fefd84